### ISSUE TITLE
Assembly error when using compositional fields with the latent heat material model.

### ISSUE THREAD

**Issue by naliboff** [geodynamics/aspect#1896]
An ASPECT user notified me of an issue with the `latent heat` material model. When one includes compositional fields the model exits with an error (`Arithmetic exception`) during the assembly of the compositional field.

The exact error is at `advection.cc lines 160:168`:
```
              data.local_rhs(i)
              += (field_term_for_rhs * scratch.phi_field[i]
                  + time_step *
                  scratch.phi_field[i]
                  * gamma
                  + scratch.phi_field[i]
                  * reaction_term)
                 *
                 JxW; 
```

Before diving into this and a rewriting a bit of the documentation in the `latent heat` material model, I wanted to check if anyone has used this model with compositional fields at some point in the recent past. Perhaps all of the models we tried so far simply set things up in an un-physical manner? Unless I missed one, none of the `latent heat` test cases or cookbooks include compositional fields. I'll add at least one new test to go with any revisions to the material model.

**Comment by jdannberg** [comment]
Now that I think about it, I have used this material model several times, but never with compositional fields. I looked into the file, and we never set the reaction terms anywhere. So if you can reproduce the issue, maybe it's just enough to add 
`for (unsigned int c=0; c<in.composition[i].size(); ++c)`
`  out.reaction_terms[i][c] = 0.0;`
to the evaluate function. 
Do we have an input file that reproduces the issue? That could be a good starting point for a test.

**Comment by naliboff** [comment]
@jdannberg - that fixed it :+1:  Thanks! I'll put in a pull request this afternoon or tomorrow with the fix and a new test case using a compositional field. I may also revise some of the wording in the material model documentation, but I've leave that discussion for the pull request.

**Comment by jdannberg** [comment]
Great! :smile:

**Comment by bangerth** [comment]
Yay signaling NaNs -- doing exactly what we wanted them to do!

**Comment by naliboff** [comment]
This issue was resolved by #1902.
